rpmlint/checks/SUIDPermissionsCheck.py

```diff
@@ -0,0 +1,212 @@
+import os
+import stat
+
+import rpm
+from rpmlint.checks.AbstractCheck import AbstractCheck
+from rpmlint.permissions import PermissionsParser, VariablesHandler
+
+
+class SUIDPermissionsCheck(AbstractCheck):
+    def __init__(self, config, output):
+        super().__init__(config, output)
+        self.perms = {}
+
+        self.var_handler = VariablesHandler('/usr/share/permissions/variables.conf')
+
+        for fname in self._paths_to('permissions', 'permissions.secure'):
+            if not os.path.exists(fname):
+                continue
+
+            self._parse_profile(fname)
+
+    def _parse_profile(self, path):
+        parser = PermissionsParser(self.var_handler, path)
+        self.perms.update(parser.getEntries())
+
+    def _is_static_entry(self, entry):
+        # entries coming from the fixed permissions profile are considered
+        # static
+        return entry.profile.endswith('/permissions')
+
+    @staticmethod
+    def _paths_to(*file_names):
+        # we used to store the permissions data in /etc even though they aren't configuration files
+        # the whitelisting should check both paths (old /etc and new /usr/share) until all
+        # distributions using the old one (SLE15) are retired
+        for name in file_names:
+            # return the new path first.
+            # chkstat prefers the new paths over the old ones, so callers that only care about the
+            # first matching file must mimic that.
+            yield '/usr/share/permissions/' + name
+            yield '/etc/' + name
+
+    def check(self, pkg):
+        if pkg.is_source:
+            return
+
+        need_set_permissions = False
+        found_suseconfig = False
+
+        for f, pkgfile in pkg.files.items():
+
+            if pkgfile.filecaps:
+                self.output.add_info(
+                    'E',
+                    pkg,
+                    'permissions-fscaps',
+                    "%(fname)s has fscaps '%(caps)s'"
+                    % {'fname': f, 'caps': pkgfile.filecaps},
+                )
+
+            mode = pkgfile.mode
+            owner = pkgfile.user + ':' + pkgfile.group
+
+            need_verifyscript = False
+            if f in self.perms or (stat.S_ISDIR(mode) and f + '/' in self.perms):
+                if stat.S_ISLNK(mode):
```

Comment:
Please move this `if` body into a separate member function. Something like `_verifyEntry(self, path)`. The `if` clause could also be a small helper function like `_existsPermissionsEntry(self, path)`.